go/adbc/driver/driverbase/connection.go

```diff
@@ -0,0 +1,403 @@
+)
+
+var (
+	infoDriverArrowVersion string
+)
+
+func init() {
+	if info, ok := debug.ReadBuildInfo(); ok {
+		for _, dep := range info.Deps {
+			switch {
+			case strings.HasPrefix(dep.Path, "github.com/apache/arrow/go/"):
+				infoDriverArrowVersion = dep.Version
+			}
+		}
+	}
+	// XXX: Deps not populated in tests
+	// https://github.com/golang/go/issues/33976
+	if infoDriverArrowVersion == "" {
+		infoDriverArrowVersion = "(unknown or development build)"
+	}
+}
+
+// ConnectionImpl is an interface that drivers implement to provide
+// vendor-specific functionality.
+type ConnectionImpl interface {
+	// adbc.ConnectionGetStatistics
+	adbc.GetSetOptions
+	Base() *ConnectionImplBase
+
+	// Will be called at most once
+	Close() error
+	// Will not be called unless autocommit is disabled
+	Commit(context.Context) error
+	CurrentCatalog() (string, bool)
+	CurrentDbSchema() (string, bool)
+	// Get boxed values for info codes
+	GetInfo(ctx context.Context, infoCodes []adbc.InfoCode) (map[adbc.InfoCode]interface{}, error)
+	// Get all info codes the driver supports (minus the 6 standard codes
+	// which are assumed to always be supported)
+	GetInfoCodes() []adbc.InfoCode
+	// Get all catalogs
+	GetObjectsCatalogs(ctx context.Context, catalog *string) ([]string, error)
+	GetObjectsDbSchemas(ctx context.Context, depth adbc.ObjectDepth, catalog *string, schema *string, metadataRecords []internal.Metadata) (map[string][]string, error)
+	GetObjectsTables(ctx context.Context, depth adbc.ObjectDepth, catalog *string, schema *string, tableName *string, columnName *string, tableType []string, metadataRecords []internal.Metadata) (map[internal.CatalogAndSchema][]internal.TableInfo, error)
+	GetTableSchema(ctx context.Context, catalog, dbSchema *string, tableName string) (*arrow.Schema, error)
+	GetTableTypes(ctx context.Context) (array.RecordReader, error)
+	NewStatement() (adbc.Statement, error)
+	ReadPartition(ctx context.Context, serializedPartition []byte) (array.RecordReader, error)
+	// Will not be called unless autocommit is disabled
+	Rollback(context.Context) error
+	SetAutocommit(enabled bool) error
+}
+
+// ConnectionImplBase is a struct that provides default implementations of the
+// ConnectionImpl interface. It is meant to be used as a composite struct for a
+// driver's ConnectionImpl implementation.
+type ConnectionImplBase struct {
+	Alloc       memory.Allocator
+	ErrorHelper ErrorHelper
+
+	Autocommit bool
+	Closed     bool
+}
+
+// NewConnectionImplBase instantiates ConnectionImplBase.
+func NewConnectionImplBase(database *DatabaseImplBase) ConnectionImplBase {
+	return ConnectionImplBase{
+		Alloc:       database.Alloc,
+		ErrorHelper: database.ErrorHelper,
+		Autocommit:  true,
+		Closed:      false,
+	}
+}
+
+func (base *ConnectionImplBase) Base() *ConnectionImplBase {
+	return base
+}
+
+func (base *ConnectionImplBase) Close() error {
+	return nil
+}
+
+func (base *ConnectionImplBase) Commit() error {
+	return nil
+}
```

Comment:
Rather than implementing these, why not leave them unimplemented forcing anything that uses `ConnectionImplBase` to still need to implement them, so we don't accidentally forget them?

Reply:
See discussion in other PR - maybe we can panic (for Close) or return an error (for Commit) instead so that it satisfies the interface by default but obviously blows up?